ddtrace/tracer/tracer.go

```diff
@@ -300,6 +332,11 @@ func (t *tracer) Stop() {
 	case <-t.stopped:
 		return
 	default:
+		if t.statsd != nil {
+			t.statsd.Incr("datadog.tracer.stopped", t.statsTags(), 1)
```

Comment:
Let's do all of this stuff at the very end of `Stop`, when we really have stopped.